src/js/planting/planting-handlers.js

```diff
@@ -93,33 +88,76 @@ Planting.prototype.plant_objects_for_view = function () {
             that.plantedobjects[i].img = img;
             that.plantedobjects[i].container = div;
         }
+    };
+};
+
+var calc_direction = function calc_direction(objectPosition, mousePosition) {
+    var calc = {};
+
+    if (objectPosition < mousePosition) {
+        calc.direction = "ASC";
+        calc.width = mousePosition - objectPosition;
+    } else if (objectPosition > mousePosition) {
+        calc.direction = "DSC";
+        calc.width = objectPosition - mousePosition;
     }
+
+    return calc;
 };
 
+var isDown = false;
 Planting.prototype.rotate_object = function (e) {
+    isDown = true;
+    $('body').addClass('noselect rotate');
+
     var plantedObject = e.data.plantedObject;
-    var direction = e.data.direction;
-    var toolBoxObjects = e.data.toolBoxObjects;
+    var projections = e.data.toolBoxObjects[plantedObject.object].projections;
 
-    var projections = toolBoxObjects[plantedObject.object].projections;
+    var side_width = 20;
+    var projection;
 
-    if (direction === 'left') {
-        if (plantedObject.projection === 0) {
-            plantedObject.projection = projections.length;
-        } else {
-            plantedObject.projection -= 1;
+    buttonX = $(this).offset().left;
+    currentProjection = plantedObject.projection;
```

Comment:
O ile się nie mylę to ```buttonX``` i ```currentProjection``` nie zostały wcześniej zadeklarowane prawda? W tych warunkach to zadziała poprawnie, ale ogólnie to jak ktoś tę libkę odpali w trybie ```strict``` to sypnie błędem. 

Dodatkowo nawet jakby odpalać to w normalnym trybie to przy czymś takim jeżeli ktoś jakimś cudem będzie używać zmiennych globalnych o takich nazwach, to w tym miejscu zostałyby one nadpisane i mogłoby się wszystko wykrzaczyć. 

Dodaj ```var``` przez zmiennymi.